packages/daemon/src/context.js

```diff
@@ -2,75 +2,78 @@
 
 import { makePromiseKit } from '@endo/promise-kit';
 
-export const makeTerminatorMaker = ({
+export const makeContextMaker = ({
   controllerForFormulaIdentifier,
   provideControllerForFormulaIdentifier,
 }) => {
   /**
    * @param {string} formulaIdentifier
    */
-  const makeTerminator = formulaIdentifier => {
-    let terminating = false;
-    const { promise: terminated, resolve: resolveTerminated } =
+  const makeContext = formulaIdentifier => {
+    let done = false;
+    const { promise: cancelled, reject: rejectCancelled } =
+      /** @type {import('@endo/promise-kit').PromiseKit<never>} */ (
+        makePromiseKit()
+      );
+    const { promise: disposed, resolve: resolveDisposed } =
       /** @type {import('@endo/promise-kit').PromiseKit<void>} */ (
         makePromiseKit()
       );
 
-    /** @type {Map<string, import('./types.js').Terminator>} */
+    /** @type {Map<string, import('./types.js').Context>} */
     const dependents = new Map();
     /** @type {Array<() => void>} */
     const hooks = [];
 
-    const terminate = (prefix = '*') => {
-      if (terminating) return terminated;
-      terminating = true;
+    const cancel = (reason, prefix = '*') => {
+      if (done) return disposed;
+      done = true;
+      rejectCancelled(reason || harden(new Error('Cancelled')));
 
       console.log(`${prefix} ${formulaIdentifier}`);
 
       controllerForFormulaIdentifier.delete(formulaIdentifier);
-      for (const dependentTerminator of dependents.values()) {
-        dependentTerminator.terminate(` ${prefix}`);
+      for (const dependentContext of dependents.values()) {
+        dependentContext.cancel(reason, ` ${prefix}`);
       }
       dependents.clear();
 
-      resolveTerminated(Promise.all(hooks.map(hook => hook())).then(() => {}));
+      resolveDisposed(Promise.all(hooks.map(hook => hook())).then(() => {}));
```

Comment:
any idea what the final then is for

Reply:
It cause the promise returned by `Promise.all()` to resolve to `undefined` as opposed to the resolved values of the hook functions. Since the hooks are about creating side effects, I think swallowing their return values in this fashion is the correct behavior.

Reply:
I added a comment for this.